### DISCUSSION TITLE
Questions about shopping list in relation to "in stock" items / cook log / meal plan

### DISCUSSION THREAD

**Discussion by silberzwiebel** [TandoorRecipes/recipes#2964] [Q&A]
Hi there,

I'm trying to use the shopping list in combination with the smart-planning tool.
My use case is: Once a week I want to do almost only a single click to get a meal plan for the week and a shopping list for these meals.

I'm confused, however, how the shopping list is working and have some questions about it:

- Recipes that are in the past week (more than 7 days ago) still appear on the shopping list (even with "last days" setting set to 1)
- Recipes that were planned even more days before are still on the list. Why is that?
- I do have to manually care about the "stock items" ("vorrätig" in German), right? Logging a cooking does not delete the items from stock, correct? That would be awesome. Otherwise, I'd just never use the "in stock" function.
- What does "shift a shopping list by 4 hours" mean ("verschieben" in German)?
- Where to can I shift a shopping list else (a menu that gives me no option "shift to [empty list]")?
- Why (and how) can I name shopping lists?

Right now, I need plenty of time to understand the shopping list. I'd be faster to go to every single recipe in the planned week and just write down the ingredients. But possibly, I misunderstand the whole idea of the shopping list. Can someone give me a few hints, how to most easily do my workflow:

- once a week smart planning
- go shopping with a no-brainer shopping list (incl. check items on shopping list)
- log every cooking (only if necessary)
- repeat and start with a fresh shopping list without old recipes

Thanks in advance!
silberzwiebel

**Comment by smilerz** [comment]
Items are in your shopping list until checked complete or deleted.

Logging cooking does not change the in stock status of a food - nor will it.   Not all food are consumed from a single recipe. There is an upcoming feature that will add a context menu to the ingredients to toggle that status from within the recipe.  I was going to wait until some front-end refactoring is complete to submit it.  But if there is demand I can submit it earlier.

> What does "shift a shopping list by 4 hours" mean ("verschieben" in German)?
You can, effectively, hide items on the shopping list by X hours.  For example: you are at the grocery store, but won't go to the butcher until tomorrow, you can postpone all the meat so that it isn't cluttering up your shopping list.

> Where to can I shift a shopping list else (a menu that gives me no option "shift to [empty list]")?
I might not understand this question.  There is only a single, continuous shopping list.
